example/main.go

```diff
@@ -0,0 +1,7 @@
+// The tools module is used to keep tool dependencies separate from the main dependencies of the repo
+// For more details see: https://github.com/golang/go/issues/25922#issuecomment-1038394599
```

Comment:
I'll change this to follow https://github.com/bazel-contrib/rules_go/blob/7f6a9bf5870f2b5ffbba1615658676dcabf9edc7/docs/go/core/bzlmod.md#depending-on-tools instead

That's a better pattern for users to follow, so that keep-sorted dependencies don't get constraint-solved along with their application dependencies.

Also we can provide this go.mod file from rules_lint, so as long as users are okay with the version of keep-sorted we have, they can skip some steps.